src/preamble.js

```diff
@@ -378,21 +375,27 @@ if (ENVIRONMENT_IS_PTHREAD) {
   // etc. and can launch a pthread. Set some fake values there meanwhile to
   // catch bugs, then set the real values in applyStackValues later.
 #if ASSERTIONS || SAFE_STACK
-  STACK_MAX = STACKTOP = STACK_MAX = 0x7FFFFFFF;
+  tempDoublePtr = STACK_MAX = STACKTOP = STACK_MAX = 0x7FFFFFFF;
 #endif
 
   Module['applyStackValues'] = function(stackBase, stackTop, stackMax) {
     STACK_BASE = stackBase;
     STACKTOP = stackTop;
     STACK_MAX = stackMax;
+    tempDoublePtr = STACK_BASE;
+#if ASSERTIONS
+    assert(tempDoublePtr % 8 == 0);
+#endif
+    STACK_BASE += 8;
```

Comment:
Should we not leave the base as is?

Reply:
What the difference between this function and `establishStackSpace` ?

Again if this code is to be portable it needs to grow the stack downwards with wasm backend.

Reply:
> What the difference between this function and `establishStackSpace` ?

I don't know, this `applyStackValues` has popped into existence during my sabbatical, it looks like the same as `establishStackSpace`.